### PR TITLE
 board: stm32_min_dev: Add support for blue/black variant

### PR DESCRIPTION
**Pull request by sidcha**
This patch adds support for two variants of stm32_min_dev board configuration.

This patch reverts changes made in [c50a88] and [fa3af7] while still retaining the change.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
LGTM, thanks.

File: boards/arm/stm32_min_dev/stm32_min_dev_black_defconfig [hunk 1]
```diff
@@ -0,0 +1,37 @@
+CONFIG_ARM=y
```
Comment (erwango) [inline boards/arm/stm32_min_dev/stm32_min_dev_black_defconfig]: Miss
```
# SPDX-License-Identifier: Apache-2.0
#
```

**Review by erwango — COMMENTED** [review]
Approach looks sane.
Since difference is located in dts but minimal, have you tried to have a common dtsi file that would be included by both variants and overwritten with the differences?

File: boards/arm/stm32_min_dev/stm32_min_dev_black_defconfig (hunk 1 above)
Comment (cbsiddharth) [reply boards/arm/stm32_min_dev/stm32_min_dev_black_defconfig]: Added

**Review by erwango — APPROVED** [review]

**Comment by galak** [comment]
Looks like you need to add these platforms to the exclude lists in `tests/kernel/interrupt/testcase.yaml` & `tests/kernel/gen_isr_table/testcase.yaml`

**Comment by cbsiddharth** [comment]
@galak, shall I make that as a separate commit or should I rebase and fixup?

**Comment by galak** [comment]
> @galak, shall I make that as a separate commit or should I rebase and fixup?

a separate commit in the PR is fine.

**Comment by cbsiddharth** [comment]
> a separate commit in the PR is fine.
Done.

File: boards/arm/stm32_min_dev/stm32_min_dev.dtsi [hunk 2]
```diff
@@ -70,7 +70,3 @@
 		status = "ok";
 	};
 };
-
-&usb {
-	status = "ok";
-};
```
Comment (explora26) [inline boards/arm/stm32_min_dev/stm32_min_dev.dtsi]: Both black and blue variant support USB device, Is there any reason to remove this ?

**Pull request merged**